src/Blocks/BlockHandler.h

```diff
@@ -241,5 +241,7 @@ class cBlockHandler
 
 protected:
 
-	BLOCKTYPE m_BlockType;
+	~cBlockHandler() = default;
+
+	const BLOCKTYPE m_BlockType;
```

Comment:
No const in member variables.

Reply:
Why not?

Reply:
See this explanation of `std::launder`: https://stackoverflow.com/a/39382728 The important point is:

>Because `n` is a `const` variable, the compiler is free to assume that `u.x.n` shall always be 1.

And we can break that compiler assumption by use of placement-new in the same memory location. You might not do this directly, but something like a container might do so internally and your code starts to break in horrible ways.

Why introduce these foot-gun. Especially when, in this case, the class is only used as `constexpr` and so can't mutate anyway.